Add speculative decoding (#83)
lmstudio-ai/mlx-engine: README.md

 - [Llava-v1.6](https://model.lmstudio.ai/download/mlx-community/llava-v1.6-mistral-7b-4bit)
   - `lms get mlx-community/llava-v1.6-mistral-7b-4bit`

### Speculative Decoding Demo

> Run the `demo.py` script with an MLX text model and a compatible `--draft-model`

Reviewer: ```suggestion
Run the `demo.py` script with an MLX text generation model and a compatible `--draft-model`
```

```bash
lms get mlx-community/Qwen2.5-7B-Instruct-4bit
lms get lmstudio-community/Qwen2.5-0.5B-Instruct-MLX-8bit

> python demo.py --model /Users/matt/.cache/lm-studio/models/mlx-community/Qwen2.5-7B-Instruct-4bit --draft-model /Users/matt/.cache/lm-studio/models/lmstudio-community/Qwen2.5-0.5B-Instruct-MLX-8bit --prompt "<|im_start|>system

Reviewer: ```suggestion
python demo.py \
    --model ~/.lmstudio/models/mlx-community/Qwen2.5-7B-Instruct-4bit \
    --draft-model ~/.lmstudio/models/lmstudio-community/Qwen2.5-0.5B-Instruct-MLX-8bit \
    --prompt "<|im_start|>system"
```

Author: Good catch

You are Qwen, created by Alibaba Cloud. You are a helpful assistant.<|im_end|>
<|im_start|>user

> Write a quick sort in C++<|im_end|>

Reviewer: ```suggestion
Write a quicksort algorithm in C++<|im_end|>
```

Author: fyi just pulled this prompt directly from Awni's example in the mlx_lm pr: https://github.com/ml-explore/mlx-examples/pull/1155

but agree to change
